### PR TITLE
layout: Add calculation of border width

### PR DESCRIPTION
**Pull request by mkiael**

### CODE REVIEW

**Comment by mkiael** [comment]
Just realized that `border-style` is a shorthand. Will change so that the style of every edge is checked.

File: layout/layout.cpp [hunk 1]
```diff
@@ -199,6 +199,31 @@ void calculate_padding(LayoutBox &box, int const font_size) {
     }
 }
 
+void calculate_border(LayoutBox &box, int const font_size) {
+    std::string_view default_style = "none";
+    std::string_view default_width = "medium";
```
Comment (robinlinden) [inline layout/layout.cpp:204]: `to_px` doesn't handle named widths right now, so right now `thin` == `medium` == `thick` == `0`. Maybe default this to `3px` (the spec says that thin, medium, and thick all are implementation-defined) or something for now so things that want a border don't get 0?

**Review by robinlinden — APPROVED** [review]
Nice!

File: layout/layout.cpp (hunk 1 above)
Comment (mkiael) [reply layout/layout.cpp:204]: Yes, that's a good idea. Will fix.

**Pull request merged**